src/integer_mod_q/polynomial_ring_zq/arithmetic/add.rs

```diff
@@ -0,0 +1,274 @@
+    ///
+    /// Parameters:
+    /// - `other`: specifies the polynomial to add to `self`
+    ///
+    /// Returns the sum of both polynomials as a [`PolynomialRingZq`].
+    ///
+    /// # Example
+    /// ```
+    /// use qfall_math::integer_mod_q::PolynomialRingZq;
+    /// use qfall_math::integer_mod_q::ModulusPolynomialRingZq;
+    /// use qfall_math::integer::PolyOverZ;
+    /// use std::str::FromStr;
+    ///
+    /// let modulus = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();
+    /// let poly_a = PolyOverZ::from_str("4  -1 0 1 1").unwrap();
+    /// let a = PolynomialRingZq::from_poly_over_z_modulus_polynomial_ring_zq(&poly_a, &modulus);
+    /// let poly_b = PolyOverZ::from_str("4  2 0 3 1").unwrap();
+    /// let b = PolynomialRingZq::from_poly_over_z_modulus_polynomial_ring_zq(&poly_b, &modulus);
+    ///
+    /// let c: PolynomialRingZq = &a + &b;
+    /// let d: PolynomialRingZq = a + b;
+    /// let e: PolynomialRingZq = &c + d;
+    /// let f: PolynomialRingZq = c + &e;
+    /// ```
+    ///
+    /// # Panics ...
+    /// - ... if the moduli of both [`PolynomialRingZq`] mismatch.
+    fn add(self, other: Self) -> Self::Output {
+        self.add_safe(other).unwrap()
+    }
+}
+
+impl PolynomialRingZq {
+    /// Implements addition for two [`PolynomialRingZq`] values.
+    ///
+    /// Parameters:
+    /// - `other`: specifies the polynomial to add to `self`
+    ///
+    /// Returns the sum of both polynomials as a [`PolynomialRingZq`] or an error if the moduli
+    /// mismatch.
+    ///
+    /// # Example
+    /// ```
+    /// use qfall_math::integer_mod_q::PolynomialRingZq;
+    /// use qfall_math::integer_mod_q::ModulusPolynomialRingZq;
+    /// use qfall_math::integer::PolyOverZ;
+    /// use std::str::FromStr;
+    ///
+    /// let modulus = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();
+    /// let poly_a = PolyOverZ::from_str("4  -1 0 1 1").unwrap();
+    /// let a = PolynomialRingZq::from_poly_over_z_modulus_polynomial_ring_zq(&poly_a, &modulus);
+    /// let poly_b = PolyOverZ::from_str("4  2 0 3 1").unwrap();
+    /// let b = PolynomialRingZq::from_poly_over_z_modulus_polynomial_ring_zq(&poly_b, &modulus);
+    ///
+    /// let c: PolynomialRingZq = a.add_safe(&b).unwrap();
+    /// ```
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`MathError::MismatchingModulus`] if the moduli of
+    /// both [`PolynomialRingZq`] mismatch.
+    pub fn add_safe(&self, other: &Self) -> Result<PolynomialRingZq, MathError> {
+        if self.modulus != other.modulus {
+            return Err(MathError::MismatchingModulus(format!(
+                " Tried to add polynomial with modulus '{}' and polynomial with modulus '{}'.",
+                self.modulus, other.modulus
+            )));
+        }
+        let mut out = PolynomialRingZq::from_poly_over_z_modulus_polynomial_ring_zq(
+            &PolyOverZ::from_str("0").unwrap(),
+            &self.modulus,
+        );
```

Comment:
better use the from trait
```suggestion
        let mut out = PolynomialRingZq::from((
            &PolyOverZ::default(),
            &self.modulus,
        ));
```